### PR TITLE
Fix issues found in 19.1

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
* Move off deprecated icons
* Suppress deprecation when it is not possible to migrate
* Fix compile issues in project wizard
* Introduce IDE profiles to keep track of required plugin versions and IDE versions

## Related Issue(s)
#814 	

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt [hunk 1]
```diff
@@ -259,18 +257,4 @@ class LambdaHandlerIndexTest {
...
-
-        val extensionPointName =
-            ExtensionPointName.create<RuntimeGroupExtensionPoint<LambdaHandlerResolver>>("aws.toolkit.lambda.handlerResolver")
-
-        PlatformTestUtil.unregisterAllExtensions(extensionPointName, projectRule.fixture.testRootDisposable)
-
-        val newVersion = LambdaHandlerIndex().version
-
-        assertThat(initialVersion).isNotEqualTo(newVersion)
-    }
```
Comment (zhangzhx) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt]: Why do we remove this test?

File: build.gradle [hunk 2]
```diff
@@ -183,14 +185,15 @@ apply plugin: 'org.jetbrains.intellij'
 apply plugin: ChangeLogPlugin
 
 intellij {
-    version "IC-" + ideaVersion
+    version ideVersion("IC")
```
Comment (zhangzhx) [inline build.gradle]: Is it better to move the product code to properties file as well?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt]: The method that it used to mess with extensions have been removed, and the new methods arent in 2018.3

https://github.com/JetBrains/intellij-community/blob/master/platform/extensions/src/com/intellij/openapi/extensions/ExtensionPoint.java#L35

Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt]: i'll look again to see if I can replicate the behavior

File: build.gradle (hunk 2 above)
Comment (abrooksv) [reply build.gradle]: Product code will be tied to the `jetbrains-*` module. I.E, `jetbrains-ult` needs to use IU

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/LambdaHandlerIndexTest.kt]: Was able to re-create the test manually:
```kotlin
    @Test
    @Suppress("DEPRECATION")
    fun testVersionChangesIfExtensionsChange() {
        val initialVersion = LambdaHandlerIndex().version

        val extensionPointName = ExtensionPointName.create<RuntimeGroupExtensionPoint<LambdaHandlerResolver>>("aws.toolkit.lambda.handlerResolver")
        val extensionPoint = extensionPointName.getPoint(null)
        val extensions = extensionPoint.extensions

        Disposer.register(projectRule.fixture.testRootDisposable, Disposable {
            extensions.forEach { extensionPoint.registerExtension(it) }
        })

        extensions.forEach { extensionPoint.unregisterExtension(it) }

        val newVersion = LambdaHandlerIndex().version

        assertThat(initialVersion).isNotEqualTo(newVersion)
    }
```

**Pull request merged**